internal/xpkg/config.go

```diff
@@ -0,0 +1,94 @@
+package xpkg
+
+import (
+	"context"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1beta1"
+)
+
+const (
+	errListImageConfigs = "cannot list ImageConfigs"
+)
+
+// ConfigStore is a store for image configuration.
+type ConfigStore interface {
+	// PullSecretFor returns the pull secret configuration for a given image.
+	PullSecretFor(ctx context.Context, image string) (imageConfig string, pullSecret string, err error)
```

Comment:
```suggestion
	PullSecretFor(ctx context.Context, image string) (imageConfig, pullSecret string, err error)
```

Nit.

`imageConfigName, pullSecretName` may also make sense (here and elsewhere) for more clarity on what the string is.